ocaml/xenopsd/lib/xenops_server.ml

```diff
@@ -1580,6 +1588,27 @@ let parallel_map name ~id lst f = parallel name ~id (List.concat_map f lst)
 
 let map_or_empty f x = Option.value ~default:[] (Option.map f x)
 
+(* Given a non-empty list, returns an Atomic rather than a single item list *)
+let nonempty_serial_unpacked name ~id lst =
+  match serial name ~id lst with
+  | [x] ->
+      x
+  | [] | _ :: _ ->
+      (* this should never happen *)
+      raise
+        (Xenopsd_error
+           (Internal_error
+              "serial with non-empty list should always return a length 1 list"
+           )
+        )
```

Comment:
This function can still raise an exception, so it can be dangerous to use, instead use this
```suggestion
let serial_of name ~id at1 at2 ats =
  Serial (id, Printf.sprintf "%s VM=%s" name id, at1 :: at2 :: ats)])
```

Reply:
Much better, thanks!

Reply:
Use the `internal_error` function that takes a printf format string.